python/scannerpy/sampler.py

```diff
@@ -71,6 +98,16 @@ def strided_range(self, video, start, end, stride):
         sample = task.samples.add()
         sample.table_name = input_table_name
         sample.column_names.extend(column_names)
-        sample.rows.extend(
-            range(min(start, num_rows), min(end, num_rows), stride))
+        sample.sampling_function = "StridedRange"
+        sampler_args = self._db._metadata_types.StridedRangeSamplerArgs()
+        sampler_args.stride = stride
+        s = start
+        while s < end:
+            ws = max(0, s - warmup_size * stride)
+            e = min(s + item_size * stride, end)
+            sampler_args.warmup_starts.append(ws)
```

Comment:
I'm assuming the runtime will check if the warmup index is the same as the start index and ignore it in that case?